modules/shared/src/main/scala/org/tessellation/cutoff/LogarithmicOrdinalCutoff.scala

```diff
@@ -0,0 +1,40 @@
+package org.tessellation.cutoff
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import eu.timepit.refined.types.numeric.NonNegLong
+
+object LogarithmicOrdinalCutoff {
+  def make: OrdinalCutoff = new OrdinalCutoff {
+    def cutoff(cutoffOrdinal: SnapshotOrdinal, currentOrdinal: SnapshotOrdinal): List[SnapshotOrdinal] = {
+      val lastOrdinal = currentOrdinal.value.value
+
+      /* NOTE:
+       * To determine the necessary number of layers, Math.log10 should be used, although it necessitates a Double as its input type.
+       *   Math.log10(lastOrdinal) + 1
+       * Converting SnapshotOrdinal to Double may result in precision loss for substantial values.
+       * Given that SnapshotOrdinal is a non-negative Long, assesing the number of digits is sufficient to ascertain the number of layers.
+       */
+      val layers = lastOrdinal.toString.length
+
+      val (ordinalsToKeep, _) = (0 until layers).toList
+        .foldLeft((List.empty[Long], lastOrdinal)) {
```

Comment:
```suggestion
      val (ordinalsToKeep, _) = (0 until layers)
        .foldLeft((Set.empty[Long], lastOrdinal)) {
```
`Range` is foldable. Also, I think we can use a set for our accumulator instead of a list because we're sorting and removing dups later anyway. It would make the code more efficient and eliminate need to `distinct` later

Reply:
:+1: